language/move-prover/tests/sources/functional/strong_edges.move

```diff
@@ -0,0 +1,88 @@
+// flag: --strong_edges
+// This file consists of a series of test cases which are client functions
+// using the standard vector module.
+module TestStrongEdges {
+    use 0x1::Vector;
+
+    spec module {
+        pragma verify = true;
+    }
+
+    struct S has key {
+        x: u64
+    }
+
+    public fun glob_and_field_edges(addr: address) acquires S {
+        let s = borrow_global_mut<S>(addr);
+        s.x = 2;
+    }
+
+    spec fun glob_and_field_edges{
+        pragma opaque = true;
+        ensures global<S>(addr).x == 2;
+        aborts_if !exists<S>(addr);
+        modifies global<S>(addr);
+    }
+
+    fun loc_edge() {
+        let r = 5;
+        let r_ref = &mut r;
+        *r_ref = 6;
+        spec {
+            assert r == 6;
+        };
+    }
+
+    fun vec_edge(v: &mut vector<u64>) : u64
+    {
+        let x = *Vector::borrow(v, 0);
+        *Vector::borrow_mut(v, 0) = 7;
+        x
+    }
+    spec fun vec_edge {
+        aborts_if len(v) == 0;
+        ensures v[0] == 7;
+        ensures v[1] == old(v[1]);
+    }
+
+    struct R has copy, drop {
+        x: u64
+    }
+
+    public fun splice_edge_helper(r_ref: &mut R) : &mut u64 {
```

Comment:
What specific aspect of strong edges are you testing with this example?

Reply:
you're right... this is testing a weak edge, albeit one that can be seen in the strong edge analysis. I was looking for coverage of the strong edge analysis (as opposed to just of strong edges). The name of the file (strong_edges) is misleading. Let me know if you want me to remove this test case or change name of file.